### PR TITLE
Add App::Link support to Cloud module

### PR DESCRIPTION
**Pull request by vejmarie**
ONLY enabled if the Cloud Module is compiled into FreeCAD
ALL files must be saved into a Cloud Based storage
Add a Signal to the PropertyLink to enable support of external storage

Signed-off-by: Jean-Marie Verdun <jmverdun3@gmail.com>

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, please confirm the following:

- [ ] Branch rebased on latest master `git pull --rebase upstream master`
- [ ] Unit tests confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ] Commit message is [well-written](https://chris.beams.io/posts/git-commit/)
- [ ] Commit message includes `issue #<id>` or `fixes #<id>` where `<id>` is the [associated MantisBT](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) issue id if one exists

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.19 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=34586).

---

### CODE REVIEW

**Comment by Vanuan** [comment]
I think adding functionality as a build-time conditional is harmful. Is it possible to do the same with a runtime condition?

**Comment by vejmarie** [comment]
> I think adding functionality as a build-time conditional is harmful. Is it possible to do the same with a runtime condition?
> Is it a common practice to change the behavior according to build time options?

Hi, everything is possible. The challenge here is that some contributor are fearing that network access might jeopardize FreeCAD. But as the changes in core are related to executing a signal, that shall be ok if everybody agrees. So enabling it at build time has the advantage to "force" the user to pick up the option he wants, and by default stand with a regular behavior.

**Comment by yorikvanhavre** [comment]
Any comment, @realthunder , @wwmayer ?

**Comment by realthunder** [comment]
I quickly scan through the changes. Looks ok to me, but I also think that there is no need to do it as build time option. Just add the signal as it is.

**Comment by vejmarie** [comment]
Ok, let me do that, it is easy. I will do it by the end of the day (US timezone). I was not sure about the "policy" we might have regarding signaling into the app, that is why I staid cautious, but who knows the signal might be useful to others!

**Comment by vejmarie** [comment]
@yorikvanhavre @realthunder just made the update

**Comment by realthunder** [comment]
Looks fine to me

File: src/Mod/Cloud/CMakeLists.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
+add_compile_options("-DBUILD_CLOUD")
```
Comment (Vanuan) [inline src/Mod/Cloud/CMakeLists.txt:1]: Still required?

Comment (vejmarie) [reply src/Mod/Cloud/CMakeLists.txt]: I just removed it, you was right it is conditionally define at the upper level.

**Comment by yorikvanhavre** [comment]
Okay, waiting for the CI test to finish and I'll merge

**Comment by vejmarie** [comment]
ping @yorikvanhavre

**Comment by yorikvanhavre** [comment]
Sorry! Done now ;)

**Comment by vejmarie** [comment]
No problem thanks !

**Comment by vejmarie** [comment]
Thanks for the update and the fix.

**Pull request merged**